### DISCUSSION TITLE
SlugNormalizer not working since update to 2.0

### DISCUSSION THREAD

**Discussion by pmochine** [thephpleague/commonmark#694] [Q&A] [answered]
Hey I have updated the code to new release, but I'm just not able to use the slug_normalizer correctly...

           $config = [
            // Extension defaults are shown below
            // If you're happy with the defaults, feel free to remove them from this array
            'heading_permalink' => [
                'html_class' => 'heading-permalink',
                'id_prefix' => 'user-content',
                'insert' => 'before',
                'title' => 'Permalink',
                'symbol' => '',
                // 'slug_normalizer' => new SlugNormalizer(),
            ],
        ];

        $environment = new Environment($config);
        $environment->addExtension(new CommonMarkCoreExtension());
        $environment->addExtension(new GithubFlavoredMarkdownExtension());
        // // Adds Table Of Content function for Markdown
        $environment->addExtension(new HeadingPermalinkExtension());
        $converter = new MarkdownConverter($environment);

I even tried to use 'slug_normalizer' => ['instance' => new SlugNormalizer()],

Any idea what I could do?

**Comment by colinodell** [comment, marked as answer]
`slug_normalizer` is now a top-level configuration item. See https://commonmark.thephpleague.com/2.0/customization/slug-normalizer/ for full details on how to configure it.

In your case, the configuration should look something like this:

```
    $config = [
        'heading_permalink' => [
            'html_class' => 'heading-permalink',
            'id_prefix' => 'user-content',
            'insert' => 'before',
            'title' => 'Permalink',
            'symbol' => '',
        ],
        'slug_normalizer' => [
            'instance' => new SlugNormalizer(), // or whatever you'd like
        ],
    ];
```

Notice how `slug_normalizer` is not part of the `heading_permalink` section any more.

Hope that helps!

**Reply by pmochine** [reply]
@colinodell ahhh thank you! 🤗
